vault/datadog_checks/vault/check.py

```diff
@@ -0,0 +1,271 @@
+            super().check(_)
+
+        self.service_check(self.SERVICE_CHECK_CONNECT, self.OK, tags=self._tags)
+
+    def check_leader_v1(self, submission_queue, dynamic_tags):
+        url = self._api_url + '/sys/leader'
+        leader_data = self.access_api(url, ignore_status_codes=SYS_LEADER_DEFAULT_CODES)
+        errors = leader_data.get('errors')
+        if errors:
+            error_msg = ';'.join(errors)
+            self.log.error('Unable to fetch leader data from Vault. Reason: %s', error_msg)
+            return
+
+        is_leader = is_affirmative(leader_data.get('is_self'))
+        dynamic_tags.append(f'is_leader:{str(is_leader).lower()}')
+
+        submission_queue.append(lambda tags: self.gauge('is_leader', int(is_leader), tags=tags))
+
+        current_leader = Leader(leader_data.get('leader_address'), leader_data.get('leader_cluster_address'))
+        has_leader = any(current_leader)  # At least one address is set
+
+        if self.config.detect_leader and has_leader:
+            if self._previous_leader is None:
+                # First check run, let's set the previous leader variable.
+                self._previous_leader = current_leader
+                return
+            if self._previous_leader == current_leader:
+                # Leader hasn't changed
+                return
+            if not is_leader:
+                # Leader has changed but the monitored vault node is not the leader. Because the agent monitors
+                # each vault node in the cluster, let's use this condition to submit a single event.
+                self._previous_leader = current_leader
+                self.log.debug(
+                    'Leader changed from %s to %s but not reporting an event as the current node is not the leader.',
+                    self._previous_leader,
+                    current_leader,
+                )
+                return
+
+            if current_leader.leader_addr != self._previous_leader.leader_addr:
+                # The main leader address has changed
+                event_message = (
+                    f'Leader address changed from `{self._previous_leader.leader_addr}` to '
+                    f'`{current_leader.leader_addr}`.'
+                )
+            else:
+                # The leader_cluster_addr changed (usually happen when the leader address points to a load balancer
+                event_message = (
+                    f'Leader cluster address changed from `{self._previous_leader.leader_cluster_addr}` to '
+                    f'`{current_leader.leader_cluster_addr}`.'
+                )
+
+            self.log.debug('Leader changed from %s to %s, sending the event.', self._previous_leader, current_leader)
+            submission_queue.append(
+                lambda tags: self.event(
+                    {
+                        'timestamp': get_timestamp(),
+                        'event_type': self.EVENT_LEADER_CHANGE,
```

Comment:
Should we still submit `source_type_name` to maintain parity with v1? I noticed `vault` isn't in the [list](https://docs.datadoghq.com/integrations/faq/list-of-api-source-attribute-value/) of source attributes though.